kyo-core/shared/src/main/scala/kyo/envs.scala

```diff
@@ -3,20 +3,32 @@ package kyo
 import kyo.core.*
 
 class Envs[+V] extends Effect[Envs[V]]:
-    type Command[T] = Unit
+    type Command[T] = Tag[T]
 
 object Envs:
     private case object envs extends Envs[Any]
     private def envs[V]: Envs[V] = envs.asInstanceOf[Envs[V]]
 
-    def get[V](using Tag[Envs[V]]): V < Envs[V] =
-        envs[V].suspend[V](())
+    // Envs.get[String] would normally use a tag of Envs[String]
+    // Yet we may provide a Layer with an output of String & Int.
+    // This would usually give the handler a tag of Envs[String & Int]
+    // However, then handler.tag != kyo.tag in the handle implementation.
+    // So we have to use this type erased tag and rely on our `accepts` implementation
+    private trait EnvsErased
+    private val envsTag: Tag[EnvsErased]     = Tag[EnvsErased]
+    private def makeEnvsTag[V]: Tag[Envs[V]] = envsTag.asInstanceOf[Tag[Envs[V]]]
```

Comment:
This here was the tricky part.

Reply:
We should try to find a safer encoding later but just the fact that the effect handling still doesn't require allocations seems worth the workaround :)

Reply:
The new State parameter of the handler is brilliant! The duality between inputs to recursive functions and state is often worth exploiting.

Reply:
It also kind of reminds me of how Futures in Rust are compiled down to state machines. It really cleans things up! 👏

Reply:
yep! It's 🤯 to think that the effect handling machinery can execute quite efficiently while still being pure without any mutability